recipes/slave.rb

```diff
@@ -110,9 +110,9 @@ class Chef::Recipe
     zookeeper_server_list: zk_server_list,
     zookeeper_port: zk_port,
     zookeeper_path: zk_path,
-    logs_dir: node['mesos']['logs_dir'],
-    work_dir: node['mesos']['work_dir'],
-    isolation_type: node['mesos']['isolation_type']
+    logs_dir: node['mesos']['slave']['env']['LOGS'],
```

Comment:
The LOGS environment variable is read by the mesosphere init wrapper, and after that it generates a "--log_dir" flag from it. This is effectively bypassing the cookbook mesos options checking and relying on the init wrapper to do the work.